pyo3-derive-backend/src/pyclass.rs

```diff
@@ -15,7 +15,7 @@ use syn::{parse_quote, Expr, Token};
 /// The parsed arguments of the pyclass macro
 pub struct PyClassArgs {
     pub freelist: Option<syn::Expr>,
-    pub name: Option<syn::Expr>,
+    pub name: Option<syn::ExprPath>,
```

Comment:
I actually wonder if this should be `syn::LitStr`?

Reply:
Could easily be `LitStr` if we deprecate the other branch as you said, although it may be better not to allow arbitrary strings, so maybe `Ident` is even better.